### PR TITLE
Feature/rpc list predefined assets

### PR DESCRIPTION
**Pull request by ysfkel**
## Issue
CU-24hwzdx https://app.clickup.com/t/24hwzdx

## Description
Implements a custom RPC end point that provides a list of all predefined assets including: 
 PICA
 LAYR
 CROWD_LOAN
 KSM
  kUSD

Includes integration tests for the feature

## Checklist

- [x ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ x] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-24hwzdx Create RPC for assets in scope](https://app.clickup.com/t/24hwzdx)

**Review by andor0 — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

**Comment by hussein-aitlahcen** [comment]
@ysfkel you have unsigned commits

**Comment by hussein-aitlahcen** [comment]
@ysfkel The PR needs to be rebased as well

**Comment by hussein-aitlahcen** [comment]
@ysfkel you have unsigned commits

**Review by hussein-aitlahcen — DISMISSED** [review]

**Review by andor0 — APPROVED** [review]

File: integration-tests/runtime-tests/test/tests/assets/rpcAssetsTests.ts [hunk 1]
```diff
@@ -14,10 +15,23 @@ describe('rpc.assets Tests', function () {
...
         expect(result).to.be.a["bignumber"].that.equals('0');
     });
+
+    it('rpc.assets.listAssets Tests', async function () {
+        if (!testConfiguration.enabledTests.rpc.listAssets__success)
+            this.skip();
+        const result = await RpcAssetsTests.rpcListAssetsTest();
+        expect(result).to.have.lengthOf(5); 
+        result.every((i) => expect(i).to.have.all.keys('id','name'))
+        expect(result.map(e=>(e.id.toNumber()))).to.include.members([ 1, 2, 3, 4, 129 ]);
```
Comment (benluelo) [inline integration-tests/runtime-tests/test/tests/assets/rpcAssetsTests.ts:25]: Is there a reason we don't check against the names as well?

File: frame/assets/rpc/src/lib.rs [hunk 2]
```diff
@@ -6,10 +6,10 @@ use jsonrpc_derive::rpc;
 use sp_api::ProvideRuntimeApi;
 use sp_blockchain::HeaderBackend;
 use sp_runtime::{generic::BlockId, traits::Block as BlockT};
-use sp_std::sync::Arc;
+use sp_std::{sync::Arc, vec::Vec};
 
 #[rpc]
-pub trait AssetsApi<BlockHash, AssetId, AccountId, Balance>
+pub trait AssetsApi<BlockHash, AssetId, AccountId, Balance, Asset>
```
Comment (benluelo) [inline frame/assets/rpc/src/lib.rs:12]: Is the generic type necessary? Since we use a concrete type for `Asset`

**Review by benluelo — CHANGES_REQUESTED** [review]

File: frame/assets/rpc/src/lib.rs (hunk 2 above)
Comment (ysfkel) [reply frame/assets/rpc/src/lib.rs:12]: do you mean the AssetsApi<> generic?

Comment (benluelo) [reply frame/assets/rpc/src/lib.rs:12]: Yes, the `Assets` generic type

Comment (ysfkel) [reply frame/assets/rpc/src/lib.rs:12]: Not sure I get what you mean, Asset is a Type required by the generic AssetsApi, can you make it clear

Comment (ysfkel) [reply frame/assets/rpc/src/lib.rs]: youre right, i have pushed a new commit that removes it

**Comment by hussein-aitlahcen** [comment]
@ysfkel We need to rebase this

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by dzmitry-lahoda — DISMISSED** [review]

File: frame/assets/rpc/src/lib.rs [hunk 3]
```diff
@@ -44,10 +44,16 @@ impl<C, Block, AssetId, AccountId, Balance>
 	for Assets<C, (Block, AssetId, AccountId, Balance)>
 where
 	Block: BlockT,
+<<<<<<< HEAD
```
Comment (dzmitry-lahoda) [inline frame/assets/rpc/src/lib.rs:47]: merge

**Comment by ysfkel** [comment]
Just realized this has not been merged , I am working to resolve the CI issues

**Comment by ysfkel** [comment]
Just realized this has not been merged , I am working to resolve the CI issues

**Review by benluelo — DISMISSED** [review]

**Review by dzmitry-lahoda — DISMISSED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

File: frame/assets/runtime-api/src/lib.rs [hunk 4]
```diff
@@ -4,17 +4,20 @@
...
 use codec::Codec;
 use composable_support::rpc_helpers::SafeRpcWrapper;
-
+use composable_traits::assets::Asset;
+use sp_std::vec::Vec;
 // Here we declare the runtime API. It is implemented it the `impl` block in
 // runtime amalgamator file (the `runtime/src/lib.rs`)
 sp_api::decl_runtime_apis! {
-	// REVIEW(benluelo): Should the AssetId type parameter be removed and then just use CurencyId directly?
+	// REVIEW(benluelo): Shou ld the AssetId type parameter be removed and then just use CurencyId directly?
```
Comment (vivekvpandya) [inline frame/assets/runtime-api/src/lib.rs:12]: Looks like IDE has broke the word `Should`

**Review by vivekvpandya — APPROVED** [review]
Just curious if all .toml files space related changes is because of IDE or some other command?

File: frame/assets/runtime-api/src/lib.rs (hunk 4 above)
Comment (ysfkel) [reply frame/assets/runtime-api/src/lib.rs:12]: should be IDE

**Pull request closed without merging**